app/models/manageiq/providers/kubernetes/container_manager/refresher.rb

```diff
@@ -1,10 +1,16 @@
 module ManageIQ::Providers::Kubernetes
   class ContainerManager::Refresher < ManageIQ::Providers::BaseManager::Refresher
     include ::EmsRefresh::Refreshers::EmsRefresherMixin
+    include ::EmsRefresh::Refreshers::ContainerRefresherMixin
 
     def self.entities
-      %w(pods services replication_controllers nodes events endpoints namespaces resource_quotas limit_ranges
-         persistent_volumes persistent_volume_claims component_statuses)
+      ['pods', 'services', 'replication_controllers', 'nodes', 'events',  'endpoints', 'namespaces', 'resource_quotas',
+       'limit_ranges', 'persistent_volumes', 'persistent_volume_claims',
+       {
+         :name => 'component_statuses',
+         :optional => true
+       }
+      ]
```

Comment:
Consider:

    [{:name => 'pods'}, {:name => 'services'}, {:name => 'replication_controllers'},
     {:name => 'nodes'}, {:name => 'events'},  {:name => 'endpoints'},
     {:name => 'namespaces'}, {:name => 'resource_quotas'},
     {:name => 'component_statuses', :default => []}]

which would simplify `fetch_entities` (no need to check if it's an `Hash`), also the `default` give you the opportunity to specify a proper default per-optional entity.